extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetSchemaReader.java

```diff
@@ -146,29 +147,29 @@ public static ParquetInstructions readParquetSchema(
             return instructionsBuilder.getValue();
         };
         final ParquetMessageDefinition colDef = new ParquetMessageDefinition();
-        final Map<String, String[]> parquetColumnNameToFirstPath = new HashMap<>();
-        final Map<Integer, Long> fieldIdCount = schema.getColumns()
+        final Map<Integer, Long> topLevelFieldIdCount = schema.getFields()
                 .stream()
-                .map(ColumnDescriptor::getPrimitiveType)
                 .map(Type::getId)
                 .filter(Objects::nonNull)
                 .map(ID::intValue)
                 .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
-        if (schema.getFieldCount() != schema.getColumns().size()) {
-            throw new IllegalStateException(String.format(
-                    "Field count inconsistent with number of columns, schema.getFieldCount()=%d, schema.getColumns().size()=%d",
-                    schema.getFieldCount(), schema.getColumns().size()));
-        }
-        final Iterator<Type> fieldIt = schema.getFields().iterator();
-        final Iterator<ColumnDescriptor> columnDescriptorIterator = schema.getColumns().iterator();
-        while (fieldIt.hasNext() && columnDescriptorIterator.hasNext()) {
-            final Type fieldType = fieldIt.next();
-            final ColumnDescriptor column = columnDescriptorIterator.next();
+        int columnIx = 0;
+        final List<ColumnDescriptor> columnDescriptors = schema.getColumns();
+        for (final Type fieldType : schema.getFields()) {
+            final int numColumns = ParquetTotalColumns.of(fieldType);
+            if (numColumns > 1) {
+                // TODO(deephaven-core#871): Parquet: Support repetition level >1 and multi-column fields
+                throw new UnsupportedOperationException(
+                        String.format("Encountered unsupported multi-column field %s, has %d total columns",
+                                fieldType.getName(), numColumns));
```

Comment:
You suggested we should maybe just start skipping nested fields. I bet we could *also* choose to include them, with some weird default. Like "UnprocessedField" singleton POJOs.